### ISSUE TITLE
Localization problem in TypeTransformation while uppercasing i to İ

### ISSUE THREAD

**Issue by bilalekremharmansa** [google/closure-compiler#3490]
Hi, 

One of my colleague encountered a localization problem today. The problem happens when default locale set to Turkish. 

In `com.google.javascript.jscomp.parsing.TypeTransformation.java#nameToKeyword(String)`

`isunknown` gets converted to `Keywords.ISUNKNOWN` by using `nameToKeyword` function. Here is how it does in code;
```
private Keywords nameToKeyword(String s) {
  return Keywords.valueOf(s.toUpperCase());
}
``` 
In a environment that uses Turkish localization, `s.toUpperCase` result as `İSUNKNOWN`. That cause `NullPointerException`

**Comment by kjin** [comment]
Created Google internal issue http://b/143291237

**Comment by kjin** [comment]
Fixed in #3491
